sdk/formrecognizer/azure-ai-formrecognizer/src/FormRecognizer.md

```diff
@@ -0,0 +1,463 @@
+public PollerFlux<OperationResult, TrainedLabeledModel> beginLabeledModelTraining(String sourceUrl, String filePrefix, boolean includeSubFolders) { }
+
+// Analyze form with custom models
+public PollerFlux<OperationResult, FormResult> beginAnalyzeForm(Flux<ByteBuffer> data, String modelId, boolean includeTextDetails) { }
+public PollerFlux<OperationResult, FormResult> beginAnalyzeForm(String uploadFilePath, String modelId, boolean includeTextDetails) { }
+
+// Analyze with supervised models
+public PollerFlux<OperationResult, LabeledFormResult> beginAnalyzeLabeledForm(Flux<ByteBuffer> data, String modelId, boolean includeTextDetails) { }
+public PollerFlux<OperationResult, LabeledFormResult> beginAnalyzeLabeledForm(String uploadFilePath, String modelId, boolean includeTextDetails) { }
+
+// List custom Models
+public PagedFlux<ModelInfo> listModels() {}
+public Mono<ModelListSummary> getModelListSummary() {}
+public Mono<Response<ModelListSummary>> getModelListSummaryWithResponse() {}
+
+// Delete model
+public Mono<Void> delete(String modelId) { }
+public Mono<Response<Void>> deleteWithResponse(String modelId) { }
+
+// Analyze receipt 
+public PollerFlux<OperationResult, ExtractedReceiptResult> beginAnalyzeReceipt(Flux<ByteBuffer> data, boolean includeTextDetails) { }
+public PollerFlux<OperationResult, ExtractedReceiptResult> beginAnalyzeReceipt(String uploadFilePath, boolean includeTextDetails) { }
+
+// Analyze Layout
+public PollerFlux<OperationResult, ExtractedLayoutResult> beginAnalyzeLayout(String uploadFilePath, boolean includeTextDetails) { }
+public PollerFlux<OperationResult, ExtractedLayoutResult> beginAnalyzeLayout(Flux<ByteBuffer> data, boolean includeTextDetails) { }
+```
+
+### Models
+
+```java
+public class OperationResult {
+    private String resultId;
+}
+
+// Unsupervised Model
+public class TrainedModel {
+    private TrainedModelInfo modelInfo;
+    private List<FormRecognizerError> trainingErrors;
+    private List<TrainingInput> trainingInputResult;
+}
+
+public class TrainedModelInfo extends ModelInfo {
+    private List<FormCluster> formClusters;
+}
+
+public class FormCluster {
+    private List<String> fieldNames;
+    private int formClusterId;
+}
+
+// Unsupervised model analyze form result
+public class FormResult {
+    private List<ExtractedPage> pages ;
+    private List<RawPageExtraction> rawPageExtractions ;
+}
+
+public class ExtractedPage {
+    private List<ExtractedField> fields ;
+    private int clusterId ;
+    private int pageNumber ;
+    private List<ExtractedTable> tables ;
+}
+
+public class ExtractedField {
+    private double confidence ;
+    private ExtractedText extractedKey ;
+    private ExtractedText extractedValue ;
+    private List<String> elements; // When includeTextDetails is set to true, a list of references to the text elements constituting this key or value.
+}
+
+public class ExtractedText {
+    private List<double> boundingBox ;
```

Comment:
Is a bounding box just four (x,y) points? It seems weird to have a List<double>? Should be be List<Point> or a `Bounds` type that contains (x1, y1), (x2, y2), etc?

Reply:
Each bounding box is actually represented by eight numbers.
>"Quadrangle bounding box, with coordinates specified relative to the top-left of the original image. The eight numbers represent the four points, clockwise from the top-left corner relative to the text orientation.."